addon/adapters/debug-adapter.js

```diff
@@ -0,0 +1,269 @@
+import { run } from '@ember/runloop';
+import MegamorphicModel from '../model';
+import DataAdapter from '@ember/debug/data-adapter';
+import { capitalize, underscore } from '@ember/string';
+import { get } from '@ember/object';
+import { A } from '@ember/array';
+import { inject } from '@ember/service';
+import M3TrackedArray from '../m3-tracked-array';
+import M3ReferenceArray from '../m3-reference-array';
+
+/*
+  Extend `Ember.DataAdapter` with m3 specific code.
+
+  @class DebugAdapter
+  @extends Ember.DataAdapter
+  @private
+*/
+
+// TODO: implement getFilters/getRecordColor/getRecordFilterValues (for model state in the cache)
+// and getRecordKeywords (for search)
+export default DataAdapter.extend({
```

Comment:
Feel free to make this an es6 class.  Might be some nice cleanup but it's certainly not necessary.

```js
class DebugAdapter extends DataAdapter {
  // ...
}
```

You would need to change the `_super` calls to `super.foo()` calls.